common/searchattribute/validator.go

```diff
@@ -118,7 +122,9 @@ func (v *Validator) Validate(searchAttributes *commonpb.SearchAttributes, namesp
 				fmt.Sprintf("%s attribute can't be set in SearchAttributes", saFieldName),
 			)
 		}
-
+		if saFieldName == BuildIds {
```

Comment:
We cannot reject all the `predefined` attributes, in a similar way `system` attributes are rejected, because there are some which are considered valid in requests. (For example - `TemporalChangeVersion`)